### PR TITLE
Add iOS build phase

### PR DESCRIPTION
**Pull request by grabbou**
Issue #87 

As of now, the only valid approach for integrating Haul with existing React Native applications is to define a script phase that runs every time we build our project and overwrites `packager.sh` (used to automatically start packager on run) and `react-native-xcode.sh` (used to bundle your app).

Problem with setting env variable `CLI_PATH` is that some scripts are defined in `React.xcodeproj`. That means scripts defined there will not receive env variables set by parent project.

Thanks to regular expressions, this is rather easy.

### CODE REVIEW

**Comment by grabbou** [comment]
screen shot 2017-04-03 at 10 17 04

**Comment by grabbou** [comment]
I decided to vendor `react-native-xcode.sh` until it ships with React Native. When that happens, we should detect version and either use vendor copy or directly from `react-native`. This is to make it possible for older apps to `bundle`.

File: src/commands/init.js [hunk 1]
```diff
@@ -140,6 +142,122 @@ async function init() {
...
+    }
+  }
+
+  // Otherwise, ask for path to a file
+  if (!entry) {
+    const result = await inquirer.prompt([
+      {
+        type: 'input',
+        name: 'entry',
+        message: 'Enter the name of the .xcodeproj file',
```
Comment (satya164) [inline src/commands/init.js]: Isn't this the `path` and not `name`?

Comment (grabbou) [inline src/commands/init.js]: Yup, it's path

File: src/commands/init.js [hunk 2]
```diff
@@ -140,6 +142,122 @@ async function init() {
...
+  }
+
+  const progress = ora('Adding haul to your native build pipeline');
+
+  await sleep();
+
+  let project = fs.readFileSync(path.join(entry, 'project.pbxproj')).toString();
+
+  // Are we already integrated?
+  if (project.includes('AD0CE2C91E925489006FC317')) {
```
Comment (satya164) [inline src/commands/init.js]: Let's store this in a constant

File: src/commands/init.js [hunk 3]
```diff
@@ -140,6 +142,122 @@ async function init() {
...
+          fs.existsSync(pathToFile) && pathToFile.includes('.xcodeproj')
+            ? true
+            : `${pathToFile} is not a valid .xcodeproj`,
+      },
+    ]);
+
+    entry = path.resolve(cwd, result.entry);
+  }
+
+  const progress = ora('Adding haul to your native build pipeline');
```
Comment (satya164) [inline src/commands/init.js]: `Adding haul to your XCode build scripts`

Though I feel like we should ask before this change. Looks rather invasive and I might just want to try before integrating.

File: src/commands/init.js [hunk 4]
```diff
@@ -140,6 +142,122 @@ async function init() {
...
+
+  await sleep();
+
+  let project = fs.readFileSync(path.join(entry, 'project.pbxproj')).toString();
+
+  // Are we already integrated?
+  if (project.includes('AD0CE2C91E925489006FC317')) {
+    progress.info('Haul is already part of your build pipeline');
```
Comment (satya164) [inline src/commands/init.js]: nit: Mention XCode

File: src/commands/init.js [hunk 5]
```diff
@@ -140,6 +142,122 @@ async function init() {
...
+        `,
+    );
+  });
+
+  if (sectionsCount > 0) {
+    fs.writeFileSync(path.join(entry, 'project.pbxproj'), project);
+    progress.succeed('Added haul to your native build pipeline');
+  } else {
+    progress.fail(
+      'Failed to add Haul to your build pipeline. See docs for manual instructions.',
```
Comment (satya164) [inline src/commands/init.js]: nit: Mention XCode

Link to docs?

File: src/commands/init.js [hunk 6]
```diff
@@ -140,6 +142,122 @@ async function init() {
...
+      dedent`
+        buildPhases = (
+          AD0CE2C91E925489006FC317 /* Integrate Haul with React Native */,
+        `,
+    );
+  });
+
+  if (sectionsCount > 0) {
+    fs.writeFileSync(path.join(entry, 'project.pbxproj'), project);
+    progress.succeed('Added haul to your native build pipeline');
```
Comment (satya164) [inline src/commands/init.js]: nit: Mention XCode

File: src/commands/init.js (hunk 4 above)
Comment (satya164) [inline src/commands/init.js]: `build pipeline` sounds very serious, maybe `build scripts` is better

**Pull request merged**